locator/token_metadata.cc

```diff
@@ -237,7 +237,7 @@ class token_metadata_impl final {
     static range<dht::token> interval_to_range(boost::icl::interval<token>::interval_type i);
 
 public:
-    future<> update_topology_change_info(dc_rack_fn& get_dc_rack);
+    future<> update_topology_change_info(dc_rack_fn<gms::inet_address>& get_dc_rack);
```

Comment:
mental note: consider passing a `variant<gms::inet_address, locator::host_id>` as an alternative.
I'm not sure if that would be simpler overall, but it seems more straight forward.
Another alternative is a class instead of a function, with `get_dc_rack(gms::inet_address)` and `get_dc_rack(locator::host_id)` methods.